### PR TITLE
fix PTK check and add warning if pygments not installed

### PR DESCRIPTION
**Pull request by gforsyth**
kept getting weird errors when pygments wasn't installed in an environment but prompt_toolkit was.  this should take care of those and provide something a little more helpful.

### CODE REVIEW

File: xonsh/shell.py [hunk 1]
```diff
@@ -60,6 +61,10 @@ def __init__(self, ctx=None, shell_type=None, config=None, rc=None,
                 warn('prompt-toolkit version < v1.0.0 is not supported. '
                      'Please update prompt-toolkit. Using readline instead.')
                 shell_type = 'readline'
+            elif not HAS_PYGMENTS:
+                warn('prompt_toolkit is available but requires pygments.  '
```
Comment (scopatz) [inline xonsh/shell.py]: Prompt toolkit no longer requires pygments.  These are separate, optional dependencies now. (Though ptk will use pygments if it is available.)

File: xonsh/platform.py [hunk 2]
```diff
@@ -81,7 +81,7 @@ def pygments_version():
 @functools.lru_cache(1)
 def has_prompt_toolkit():
     """ Tests if the `prompt_toolkit` is available. """
-    spec = importlib.util.find_spec('pygments')
+    spec = importlib.util.find_spec('prompt_toolkit')
```
Comment (scopatz) [inline xonsh/platform.py]: Doh!

File: xonsh/shell.py (hunk 1 above)
Comment (gforsyth) [inline xonsh/shell.py]: Yeah, that's what got me started on this.  Should it instead warn that "xonsh requires pygments alongside prompt_toolkit"?

Comment (scopatz) [inline xonsh/shell.py]: well I would argue that we shouldn't require that, but that change might be more involved:)

File: xonsh/shell.py [hunk 3]
```diff
@@ -60,6 +61,10 @@ def __init__(self, ctx=None, shell_type=None, config=None, rc=None,
...
+                     'Please install pygments. Using readline shell')
+                shell_type = 'readline'
```
Comment (adqm) [inline xonsh/shell.py]: If we have ptk but not pygments, I think we shouldn't need to fall all the way back to readline.  I think we may be able just to turn `$COLOR_INPUT` and `$COLOR_RESULTS` off in this case, and still to use ptk.

That way, we don't have to say that we are falling back to readline, but just that syntax highlighting has been disabled, or something along those lines.

**Comment by laerus** [comment]
hey @gforsyth !

i'm getting this when only prompt_toolkit exist and no pygments:

```
$ XONSH_DEBUG=1 xonsh
/home/laerus/.venvs/xonsh/lib/python3.4/site-packages/xonsh/shell.py:65: UserWarning: pygments not found. syntax highlighting disabled.
  warn('pygments not found. syntax highlighting disabled.')
Traceback (most recent call last):
  File "/home/laerus/.venvs/xonsh/bin/xonsh", line 3, in <module>
    main()
  File "/home/laerus/.venvs/xonsh/lib/python3.4/site-packages/xonsh/main.py", line 246, in main
    shell.shell.cmdloop()
  File "/home/laerus/.venvs/xonsh/lib/python3.4/site-packages/xonsh/ptk/shell.py", line 130, in cmdloop
    line = self.singleline(auto_suggest=auto_suggest)
  File "/home/laerus/.venvs/xonsh/lib/python3.4/site-packages/xonsh/ptk/shell.py", line 77, in singleline
    rprompt_tokens = self.rprompt_tokens(None)
  File "/home/laerus/.venvs/xonsh/lib/python3.4/site-packages/xonsh/ptk/shell.py", line 169, in rprompt_tokens
    toks = pyghooks.partial_color_tokenize(p)
NameError: name 'pyghooks' is not defined
```

**Comment by scopatz** [comment]
This might need to be merged with master as well.

**Comment by gforsyth** [comment]
Hey @laerus -- I honestly can't even remember what I was doing here -- I'm going to abandon this and start fresh.

**Pull request closed without merging**